File: scenarios/app-configuration-typescript-config-values/golden/src/app-configuration.ts
import {
  AppConfigurationClient,
  featureFlagContentType,
  featureFlagPrefix,
  type ConfigurationSetting,
  type FeatureFlagValue,
} from "@azure/app-configuration";

declare const process: {
  env: Record<string, string | undefined>;
  exitCode?: number;
};

function requireEnvironment(name: string): string {
  const value = process.env[name];
  if (!value) {
    throw new Error(`Set ${name} before running.`);
  }
  return value;
}

function isRestError(error: unknown): error is Error & { statusCode: number } {
  return (
    error instanceof Error &&
    "statusCode" in error &&
    typeof error.statusCode === "number"
  );
}

async function main(): Promise<void> {
  const client = new AppConfigurationClient(
    requireEnvironment("AZURE_APPCONFIG_CONNECTION_STRING"),
  );
  const key = "app:Settings:FontSize";

  try {
    await client.setConfigurationSetting({ key, value: "24" });
    await client.setConfigurationSetting({
      key,
      value: "24",
      label: "Production",
    });

    const setting = await client.getConfigurationSetting({ key });
    console.log(setting.value);

    for await (const item of client.listConfigurationSettings({
      keyFilter: "app:Settings:*",
    })) {
      console.log(`${item.key}=${item.value}`);
    }

    const featureFlag: ConfigurationSetting<FeatureFlagValue> = {
      key: `${featureFlagPrefix}BetaFeature`,
      contentType: featureFlagContentType,
      isReadOnly: false,
      value: {
        id: "BetaFeature",
        enabled: true,
        conditions: { clientFilters: [] },
      },
    };
    await client.setConfigurationSetting(featureFlag);
    await client.deleteConfigurationSetting({ key });
  } catch (error: unknown) {
    if (isRestError(error)) {
      console.error(
        `App Configuration request failed with status ${error.statusCode}.`,
      );
      process.exitCode = 1;
    } else {
      throw error;
    }
  }
}

await main();

File: scenarios/app-configuration-python-config-values/fixtures/baseline-33420505368/app_configuration.py
import os
import sys

from azure.appconfiguration import (
    AzureAppConfigurationClient,
    FeatureFlagConfigurationSetting,
)
from azure.core.exceptions import HttpResponseError
from azure.identity import DefaultAzureCredential


SETTING_KEY = "app:Settings:FontSize"


def main() -> int:
    endpoint = os.environ.get("AZURE_APPCONFIG_ENDPOINT")
    if not endpoint:
        print(
            "Set AZURE_APPCONFIG_ENDPOINT to the App Configuration endpoint.",
            file=sys.stderr,
        )
        return 2

    try:
        with DefaultAzureCredential() as credential:
            with AzureAppConfigurationClient(
                base_url=endpoint,
                credential=credential,
            ) as client:
                client.set_configuration_setting(key=SETTING_KEY, value="24")
                client.set_configuration_setting(
                    key=SETTING_KEY,
                    value="24",
                    label="Production",
                )

                setting = client.get_configuration_setting(key=SETTING_KEY)
                print(setting.value)

                for matching_setting in client.list_configuration_settings(
                    key_filter="app:Settings:*"
                ):
                    print(
                        f"{matching_setting.key}={matching_setting.value}"
                        f" (label={matching_setting.label!r})"
                    )

                client.set_configuration_setting(
                    FeatureFlagConfigurationSetting(
                        feature_id="BetaFeature",
                        enabled=True,
                    )
                )

                client.delete_configuration_setting(key=SETTING_KEY)
    except HttpResponseError as error:
        print(f"Azure App Configuration request failed: {error}", file=sys.stderr)
        return 1

    return 0


if __name__ == "__main__":
    raise SystemExit(main())
